application/plugins/stop_manager/stop_manager.php

```diff
@@ -0,0 +1,268 @@
+    $type_keywords = array_map('strtolower', $config['type_keywords']);
+
+    // A la reception du message, si c'est un message STOP (STOP rappel) par exemple
+    // Le mettre dans la table des STOP
+    $msg = $sms->TextDecoded;
+    $from = $sms->SenderNumber;
+    //$msg_user = $sms->msg_user;
+
+    $cmds_valides = array_merge($optout_keywords, $optin_keywords);
+    $types_valides = $type_keywords;
+
+    $types_reg = implode('|', $types_valides);
+    $cmds_reg = implode('|', $cmds_valides);
+
+    if ($config['enable_type'])
+        $ret = preg_match('/\b('.$cmds_reg.')\s*('.$types_reg.')\b/i', $msg, $matches, PREG_UNMATCHED_AS_NULL);
+    else
+        $ret = preg_match('/\b('.$cmds_reg.')\b/i', $msg, $matches, PREG_UNMATCHED_AS_NULL);
+
+    if ($ret) {
+        $cmd = strtoupper($matches[1]);
+        $type = ($config['enable_type']) ? strtolower($matches[2]) : "TYPE_NOT_SET_SO_STOP_ALL";
+        $CI =& get_instance();
+        $CI->load->model('stop_manager/Stop_manager_model', 'Stop_manager_model');
+
+        $text = "";
+
+        //var_dump($matches);
+        switch (true) {
+            case in_array($cmd, $optout_keywords):
+                $ret = $CI->Stop_manager_model->add($from, $type, $msg);
+
+                $strTemplate = ':received_command pris en compte.';
+                if ($config['enable_optin'])
+                    $strTemplate .= ' Pour recevoir à nouveau, répondre ":optin_command"';
+
+                $strParams = [
+                    ':received_command' => ($config['enable_type']) ? $cmd.' '.$type : $cmd,
+                    ':optin_command' => ($config['enable_type']) ? $optin_keywords[0].' '.$type : $optin_keywords[0],
+                ];
+
+                $text = strtr($strTemplate, $strParams);
+                if ($config['enable_autoreply_info'])
+                    autoreply($from, $text);
+                break;
+            case (in_array($cmd, $optin_keywords) && $config['enable_optin']) :
+                $ret = $CI->Stop_manager_model->delete($from, $type);
+
+                $strTemplate = ':received_command pris en compte.';
+                $strTemplate .= ' Pour ne plus recevoir, répondre ":optout_command"';
+
+                $strParams = [
+                    ':received_command' => ($config['enable_type']) ? $cmd.' '.$type : $cmd,
+                    ':optout_command' => ($config['enable_type']) ? $optout_keywords[0].' '.$type : $optout_keywords[0],
+                ];
+
+                $text = strtr($strTemplate, $strParams);
+                if ($config['enable_autoreply_info'])
+                    autoreply($from, $text);
+                break;
+            default:
+                $text = "Demande non valide ($msg)";
+                if ($config['enable_autoreply_error'])
+                    autoreply($from, $text);
+                break;
+        }
+    } else {
+        $strTemplate = "Demande non valide (:received_command). Répondre ':optout_keyword";
```

Comment:
please move localisation to its respective files.

Reply:
What does "tni_" stand for in the kalkun_lang.php files ? I'm looking for a place to put the translations and was wondering is there is anything specific to plugins